CMakeLists.txt

```diff
@@ -32,6 +32,47 @@ if(EXISTS "${CMAKE_SOURCE_DIR}/.git")
 endif()
 
 
+# Search for the path to dir called "readline" using find_path.  If
+# found, include this via include_directories() below.  This ends up
+# as -I<PATH> in the generated files for make.
+find_path(READLINE_INCLUDE_DIR readline
+          HINTS ${READLINE_ROOT_DIR} PATH_SUFFIXES usr usr/include)
+
+# Search for the dir where library "readline" may be found.  The name
+# is translated to platform-specific library file name in
+# find_library.  If found, add that path via link_directories() below.
+# This ends up as -L<PATH> in the generated files for make.
+find_library(READLINE_LIB NAMES readline
+  HINTS ${READLINE_ROOT_DIR} usr PATH_SUFFIXES lib lib64)
+
```

Comment:
```cmake
include(FindPackageHandleStandardArgs)
find_package_handle_standard_args(Readline
                                  REQUIRED_VARS READLINE_INCLUDE_DIR READLINE_LIBRARY)
```
So `find_path` and `find_library` are manual means of finding the library they expect variables to be populated. I found [this](https://cmake.org/cmake/help/latest/module/FindPackageHandleStandardArgs.html) is a programmatic way to find package and populate the variables ones wants. This however also relies on cmake based package finding for GNU packages `pkgConfig` is the suggested way.